driver/device.cpp

```diff
@@ -304,6 +313,27 @@ void Device::connect_endpoint_(DeviceEndpointInfo& endpoint_info)
     net_transceiver_->connect(endpoint_info);
 }
 
+
+bool Device::remove_slot_endpoint_(roc_slot slot)
+{
+    for (unsigned i=0; i<info_.remote_endpoints.size(); ++i){
+        if (info_.remote_endpoints[i].slot == slot) {
+            info_.remote_endpoints.erase(info_.remote_endpoints.begin() + i);
+            return true;
+        }
+    }
+    return false;
```

Comment:
A few minor suggestions:

1. Inline `remove_slot_endpoint_()` into `unlink()`. Manipulations with endpoint vectors are done directly in `bind()` and `connect()`, so it would be easier to follow if `unlink()` behaves similarly.

2. Cover both `remote_endpoints` (updated by connect) and `local_endpoints` (updated by bind)

3. Use std::remove_if:

```suggestion
if (auto new_end = std::remove_if(info_.remote_endpoints.begin(),
             info_.remote_endpoints.end(),
             [slot](const auto& endpoint) { return endpoint.slot == slot; });
    new_end != info_.remote_endpoints.end()) {
    info_.remote_endpoints.erase(new_end, info_.remote_endpoints.end());
}

// same for local_endpoints
```